data_steward/utils/bq.py

```diff
@@ -286,3 +286,64 @@ def get_latest_validation_dataset_id(project_id):
             return sorted(validation_datasets, key=lambda x: x[0],
                           reverse=True)[0][1]
     return None
+
+
+def create_dataset(project_id,
+                   dataset_id,
+                   description=None,
+                   friendly_name=None,
+                   overwrite_existing=None):
+    """
+    Creates a new dataset
+
+    :param project_id: name of the project to create dataset in, defaults to the currently configured project if missing.
+    :param dataset_id: name to give the new dataset - required
+    :param description: dataset description - required
+    :param friendly_name: user friendly name for dataset - optional
+    :param overwrite_existing: determine if dataset should be overwritten if already exists, defaults to true/overwrite
+    :return: a new dataset returned from the API
+    """
+    if dataset_id is None:
+        raise RuntimeError("Cannot create a dataset without a name")
+
+    if description is None:
+        raise RuntimeError("Will not create a dataset without a description")
+
+    if project_id is None:
+        LOGGER.info(f"You should specify project_id for a reliable experience."
+                    f"Defaulting to {os.environ.get(PROJECT_ID)}.")
+        project_id = os.environ.get(PROJECT_ID)
+
+    if overwrite_existing is None:
+        overwrite_existing = True
+    elif not overwrite_existing:
+        overwrite_existing = False
+    else:
+        overwrite_existing = True
```

Comment:
The `None` is what was in the `bq_utils.create_dataset`. It defaults to None, if it is not provided, and overwrites; per the description `:param overwrite_existing: determine if dataset should be overwritten if already exists, defaults to true/overwrite` Let me know if this needs to be changed. I just didn't want to alter want was already in place.

Reply:
I think it makes sense to change it from `None` to `True`. We shouldn't have to stick to legacy conventions in new code, but thanks for the reasoning on why it was added. When we make the change from `bq_utils` to `bq`, we need to make sure all usages conform to the new convention.

Reply:
Changing it to `True` will then default to `False` if the overwrite_existing parameter is not given. The currently logic I think is just wonky to being with. Both `None` and `not overwrite_existing` should both default to `True` if we want to follow what it says in the notes. I'm thinking it would make the most sense to just change the logic to the following:

```
if overwrite_existing is False:
   overwrite_existing = False
else:
   overwrite_existing = True
```

that way both `True` and None will default to `True`

Reply:
This snippet of code is not really changing much. I think an error should be raised if `overwrite_existing` is neither `True` nor `False` (the value in the function definition should be `True` as well). If it is boolean, there are no issues and the `exists_ok=overwrite_existing` parameter setting should handle those cases.

Reply:
yeah that makes sense to me, and with `sandbox.create_sandbox_dataset` it is always declaring that parameter as `False`. So in this case do I just remove that if clause entirely?

Reply:
Yeah removing is best.